hack/generated/controllers/generic_controller.go

```diff
@@ -0,0 +1,749 @@
+
+	controllerutil.AddFinalizer(metaObj, GenericControllerFinalizer)
+
+	sig, err := SpecSignature(metaObj)
+	if err != nil {
+		return errors.Wrap(err, "failed to compute resource spec hash")
+	}
+
+	annotations := make(map[string]string)
+	annotations[ResourceSigAnnotationKey] = sig
+	annotations[DeploymentIdAnnotation] = deployment.Id
+	annotations[DeploymentNameAnnotation] = deployment.Name
+
+	// TODO: Do we want to just use Azure's annotations here? I bet we don't? We probably want to map
+	// TODO: them onto something more robust? For now just use Azure's though.
+	annotations[ResourceStateAnnotation] = string(deployment.Properties.ProvisioningState)
+
+	if deployment.IsTerminalProvisioningState() {
+		if deployment.Properties.ProvisioningState == armclient.FailedProvisioningState {
+			annotations[ResourceErrorAnnotation] = deployment.Properties.Error.String()
+		} else if len(deployment.Properties.OutputResources) == 0 {
+			return errors.Errorf("Template deployment didn't have any output resources")
+		} else {
+			resourceId := deployment.Properties.OutputResources[0].ID
+			annotations[ResourceIdAnnotation] = resourceId
+
+			if status != nil {
+				err = SetStatus(metaObj, status)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	addAnnotations(metaObj, annotations)
+	return nil
+}
+
+func (gr *GenericReconciler) createDeployment(ctx context.Context, data *ReconcileMetadata) (ctrl.Result, error) {
+	deployment, err := gr.resourceSpecToDeployment(data)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if err := patcher(ctx, gr.Client, data.metaObj, func(ctx context.Context, mutObj genruntime.MetaObject) error {
+		deployment, err = gr.ARMClient.CreateDeployment(ctx, deployment)
+		if err != nil {
+			return err
+		}
+
+		err = gr.updateMetaObject(mutObj, deployment, nil) // Status is always nil here
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}); err != nil {
+		// This is a superfluous error as per https://github.com/kubernetes-sigs/controller-runtime/issues/377
+		// The correct handling is just to ignore it and we will get an event shortly with the updated version to patch
+		if apierrors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+
+		return ctrl.Result{}, errors.Wrap(err, "failed to patch")
+	}
+
+	result := ctrl.Result{}
+	// TODO: This is going to be common... need a wrapper/helper somehow?
+	if !deployment.IsTerminalProvisioningState() {
+		result = ctrl.Result{
+			RequeueAfter: 5 * time.Second,
+		}
+	}
+	return result, err
+}
+
+// TODO: There's a bit too much duplicated code between this and create deployment -- should be a good way to combine them?
+func (gr *GenericReconciler) watchDeployment(ctx context.Context, data *ReconcileMetadata) (ctrl.Result, error) {
+	deployment, err := gr.resourceSpecToDeployment(data)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if err := patcher(ctx, gr.Client, data.metaObj, func(ctx context.Context, mutObj genruntime.MetaObject) error {
```

Comment:
Suggest extracting this **func** out into a separate method; at least then all of the condition for the **if** will fit on one line.

Reply:
Done

Reply:
Fixed